### ISSUE TITLE
Typescript type errors in strict mode

### ISSUE THREAD

**Issue by caseylucas** [jacobwgillespie/styled-icons#607]
When using styled-icons with create-react-app (although I doubt that matters) and typescript configured in "strict" mode (the default for create-react-app), an error is generated when performing a production build.

All of the generated *.tsx files have the same error. An example:

```
.../node_modules/styled-icons/fa-brands/Wpforms/Wpforms.tsx
Type error: Element implicitly has an 'any' type because type '{}' has no index signature.  TS7017

    22 |     (p, k) => {
    23 |       if (validProp(k)) {
  > 24 |         p[k] = otherProps[k]
       |         ^
    25 |       }
    26 |       return p
    27 |     }, {}
```

This error can be fixed by casting the empty object on line 27:
```
    }, {} as {[x:string]: typeof otherProps[keyof typeof otherProps]}
```

But then another error occurs:

```
Type error: Element implicitly has an 'any' type because type '{ 'aria-hidden'?: string | undefined; className?: string | undefined; color?: string | undefined; height?: string | number | undefined; id?: string | undefined; lang?: string | undefined; ... 412 more ...; key?: string | ... 1 more ... | undefined; }' has no index signature.  TS7017

    22 |     (p, k) => {
    23 |       if (validProp(k)) {
  > 24 |         p[k] = otherProps[k]
       |                ^
    25 |       }
    26 |       return p
    27 |     }, {} as {[x:string]: typeof otherProps[keyof typeof otherProps]}
```

This error can be fixed by helping out the return value from Object.keys. Changed from:

```
  const svgProps = Object.keys(otherProps).reduce(
```
to
```
  const svgProps = (Object.keys(otherProps) as Array<keyof typeof otherProps>).reduce(
```
Another option is to add `// @ts-ignore` before line 24.

**Comment by jacobwgillespie** [comment]
Thanks for this!  I think partially the issue is that the `.tsx` files are accidentally ending up inside the bundle, so I should get those removed, but this should allow me to build with `"strict": true` inside this repo, which apparently wasn't already the case.

**Comment by jacobwgillespie** [comment]
This has been released as v7.0.1 - thanks again!
